### DISCUSSION TITLE
Upload from url

### DISCUSSION THREAD

**Discussion by DearFox** [shish/shimmie2#1278] [Q&A] [answered]
I still don’t understand if I can publish a post using a link to a picture rather than attaching a file.
Sorry for the possibly stupid question, but I couldn't find the answer after an hour of searching...

**Comment by shish** [comment, marked as answer]
In `Board Config` -> `Upload` there should be a `Transload` setting with one of several options - disabled by default, and then curl / wget / fopen depending on which things your web host has installed. If you have one of the latter three active, then on the `/upload` page you should see a bunch of URL textboxes alongside the "pick a file to upload" buttons.

**Reply by DearFox** [reply]
Thank you very much for your help.

Also, if anyone is interested: you can publish posts using `http://<domain of your site>/upload?url=<link to picture>`
I found out about this a little by accident while messing around with [Upload to Danbooru](https://addons.mozilla.org/ru/firefox/addon/upload-to-danbooru/). By default it won't work with shimmie2 but you can edit `uploadURLGenerator.js` to make it work with shimmie2 which is quite convenient.

**Reply by W01360** [reply]
can you please tell us the process for how to edit the file .js? (sorry for my english)
